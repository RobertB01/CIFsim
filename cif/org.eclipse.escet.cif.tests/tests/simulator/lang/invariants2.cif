//////////////////////////////////////////////////////////////////////////////
// Copyright (c) 2010, 2021 Contributors to the Eclipse Foundation
//
// See the NOTICE file(s) distributed with this work for additional
// information regarding copyright ownership.
//
// This program and the accompanying materials are made available
// under the terms of the MIT License which is available at
// https://opensource.org/licenses/MIT
//
// SPDX-License-Identifier: MIT
//////////////////////////////////////////////////////////////////////////////

event e1, e2;

automaton p:
  disc int x = 0;

  location l1:
    initial;
    invariant x < 2;

    edge e1 do x := x + 1;
    edge e2 goto l2;

  location l2:
    urgent;
    invariant x < 4;

    edge do x := x + 1;
end
