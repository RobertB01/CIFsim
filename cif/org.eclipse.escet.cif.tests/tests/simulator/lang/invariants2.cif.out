Initial state: time=0.0, p=l1, p.x=0

Possible transitions:
  #1: event e1
  #2: event e2

Transition: event e1
State: time=0.0, p=l1, p.x=1

Transition: event e2
State: time=0.0, p=l2, p.x=1

Transition: event tau
State: time=0.0, p=l2, p.x=2

Transition: event tau
State: time=0.0, p=l2, p.x=3

Simulation resulted in deadlock.
